avro-fastserde/src/test/java/com/linkedin/avro/fastserde/generated/avro/AnotherSubRecord.java

```diff
@@ -0,0 +1,282 @@
+/**
+ * Autogenerated by Avro
```

Comment:
What's with all the autogenerated code being committed?

Reply:
What is the recommended way for those auto-generated classes in this repo?
Without those classes, 'gradle build` will fail right away.
I understand that we should trigger the command to generate those classes first then run 'gradle build', but it seems not so convenient.

Reply:
I guess these are two build management philosophies.

Approach 1 is to check in the generated classes, such that if someone clones the repo and opens it up in their IDE, it works right away (i.e.: no red lines for classes missing). This is the approach Gaojie is following in this PR.

Approach 2 is to not check in the generated classes, such that if someone clones the repo, they have to run the build at least once, so that the build auto-generates the classes, prior to the project looking fine in the IDE.

Based on @smccauliff's comment, it seems like this project might prefer following approach 2. As long as there is a good way in Gradle to generate Avro classes for multiple Avro versions, then that is fine. I guess, worst case scenario, in the absence of a clean Gradle way, we could make Gradle shell out to the `regenerate_avro.sh` script... If we did that, we would probably want to add the generated classes to `.gitignore`.

Reply:
im working on better codegen support (See avro-codegen module) that would allow generating the java code out of avro schemas (via a utility 1st then a gradle plugin).

i believe once that exists, and incombination with the idea plugin (https://docs.gradle.org/current/userguide/idea_plugin.html) it would be possible to define a codegen goal and make the idea task depend on the codegen one.

IIUD that would make intellij run the codegen when opening the project, and a similar thing could be done for builds in general.

however, it may be too soon to do it now

Reply:
Removed the generated classes from the commit